diff --git a/src/Exporters/ExportEPUB.cpp b/src/Exporters/ExportEPUB.cpp
index 1ff1055d1..00d1d86eb 100644
--- a/src/Exporters/ExportEPUB.cpp
+++ b/src/Exporters/ExportEPUB.cpp
@@ -25,9 +25,11 @@
 #endif
 
 #include <string>
+#include <stdint.h>
 #include <string.h>
 
 #include <zip.h>
+#include <zlib.h>
 #ifdef _WIN32
 #include <iowin32.h>
 #endif
@@ -145,7 +147,7 @@ void ExportEPUB::SaveFolderAsEpubToLocation(const QString &fullfolderpath, const
     fileInfo.tmz_date.tm_year = timeNow.date().year();
 
      // Write the mimetype. This must be uncompressed and the first entry in the archive.
-    if (zipOpenNewFileInZip64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 0, 0) != ZIP_OK) {
+    if (zipOpenNewFileInZip4_64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 8, 0, 15, 8, Z_DEFAULT_STRATEGY, NULL, 0, 0x0b00, 1<<11, 0) != ZIP_OK) {
         zipClose(zfile, NULL);
         QFile::remove(tempFile);
         throw(CannotStoreFile("mimetype"));
diff --git a/src/Importers/ImportEPUB.cpp b/src/Importers/ImportEPUB.cpp
index a2020af0c..8dd247e2e 100644
--- a/src/Importers/ImportEPUB.cpp
+++ b/src/Importers/ImportEPUB.cpp
@@ -25,6 +25,8 @@
 #define NOMINMAX
 #endif
 
+#include <stdint.h>
+
 #include "unzip.h"
 #ifdef _WIN32
 #include "iowin32.h"
diff --git a/src/Misc/Utility.cpp b/src/Misc/Utility.cpp
index 413b0bd85..a13a6aef2 100644
--- a/src/Misc/Utility.cpp
+++ b/src/Misc/Utility.cpp
@@ -25,6 +25,8 @@
 #define NOMINMAX
 #endif
 
+#include <stdint.h>
+
 #include "unzip.h"
 
 #ifdef _WIN32
