diff -up Sigil-0.9.13/src/Exporters/ExportEPUB.cpp.mz Sigil-0.9.13/src/Exporters/ExportEPUB.cpp
--- Sigil-0.9.13/src/Exporters/ExportEPUB.cpp.mz	2019-03-20 01:01:44.000000000 +0100
+++ Sigil-0.9.13/src/Exporters/ExportEPUB.cpp	2019-06-04 16:00:33.288040415 +0200
@@ -25,7 +25,9 @@
 
 #include <string>
 #include <string.h>
+#include <stdint.h>
 #include <zip.h>
+#include <zlib.h>
 #ifdef _WIN32
 #include <iowin32.h>
 #endif
@@ -143,7 +145,7 @@ void ExportEPUB::SaveFolderAsEpubToLocat
     fileInfo.tmz_date.tm_year = timeNow.date().year();
 
     // Write the mimetype. This must be uncompressed and the first entry in the archive.
-    if (zipOpenNewFileInZip64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 0, 0) != Z_OK) {
+    if (zipOpenNewFileInZip4_64(zfile, "mimetype", &fileInfo, NULL, 0, NULL, 0, NULL, Z_NO_COMPRESSION, 8, 0, 15, 8, Z_DEFAULT_STRATEGY, NULL, 0, 0x0b00, 1<<11, 0) != Z_OK) {
         zipClose(zfile, NULL);
         QFile::remove(tempFile);
         throw(CannotStoreFile("mimetype"));
diff -up Sigil-0.9.13/src/Importers/ImportEPUB.cpp.mz Sigil-0.9.13/src/Importers/ImportEPUB.cpp
--- Sigil-0.9.13/src/Importers/ImportEPUB.cpp.mz	2019-03-20 01:01:44.000000000 +0100
+++ Sigil-0.9.13/src/Importers/ImportEPUB.cpp	2019-06-04 16:01:06.433908894 +0200
@@ -25,6 +25,7 @@
 #define NOMINMAX
 #endif
 
+#include <stdint.h>
 #include "unzip.h"
 #ifdef _WIN32
 #include "iowin32.h"
diff -up Sigil-0.9.13/src/Misc/Utility.cpp.mz Sigil-0.9.13/src/Misc/Utility.cpp
--- Sigil-0.9.13/src/Misc/Utility.cpp.mz	2019-03-20 01:01:44.000000000 +0100
+++ Sigil-0.9.13/src/Misc/Utility.cpp	2019-06-04 16:01:22.951850121 +0200
@@ -23,6 +23,7 @@
 #define NOMINMAX
 #endif
 
+#include <stdint.h>
 #include "unzip.h"
 #ifdef _WIN32
 #include "iowin32.h"
